### PR TITLE
feat(positionning): open popperOptions api

### PR DESCRIPTION
**Pull request by fbasso**
First draft about how to give access to the popper options.

### CODE REVIEW

File: src/datepicker/datepicker-input-config.ts [hunk 1]
```diff
@@ -17,5 +18,6 @@ export class NgbInputDatepickerConfig extends NgbDatepickerConfig {
   container: null | 'body';
   positionTarget: string | HTMLElement;
   placement: PlacementArray = ['bottom-start', 'bottom-end', 'top-start', 'top-end'];
+  popperOptions: (options: Partial<Options>) => Partial<Options>;
```
Comment (maxokorokov) [inline src/datepicker/datepicker-input-config.ts:21]: maybe we should initialize directly with function, not undefined?

File: src/datepicker/datepicker-input.ts [hunk 2]
```diff
@@ -289,6 +298,7 @@ export class NgbInputDatepicker implements OnChanges,
       @Inject(DOCUMENT) private _document: any, private _changeDetector: ChangeDetectorRef,
       config: NgbInputDatepickerConfig) {
     ['autoClose', 'container', 'positionTarget', 'placement'].forEach(input => this[input] = config[input]);
+    this.popperOptions = config.popperOptions || ((options: Partial<Options>) => options);
```
Comment (maxokorokov) [inline src/datepicker/datepicker-input.ts:301]: if we initialize directly in config, we should avoid having this

File: src/datepicker/datepicker-input.ts [hunk 3]
```diff
@@ -394,7 +404,9 @@ export class NgbInputDatepicker implements OnChanges,
             targetElement: this._cRef.location.nativeElement,
             placement: this.placement,
             appendToBody: this.container === 'body',
-            updatePopperOptions: addPopperOffset([0, 2])
+            updatePopperOptions: (options) => {
```
Comment (maxokorokov) [inline src/datepicker/datepicker-input.ts:407]: nitpick: arrow function?

File: src/util/positioning.spec.ts [hunk 4]
```diff
@@ -55,11 +55,6 @@ describe('positioning', () => {
   it('should convert popper classes to bootstrap classes', () => {
 
     for (const[popperClass, bsClass] of Object.entries(matchingPopperBootstrapPlacements)) {
-      if (bsClass !== getBootstrapBaseClassPlacement('', popperClass as PopperPlacement)) {
```
Comment (maxokorokov) [inline src/util/positioning.spec.ts:58]: 👍

File: src/popover/popover.spec.ts [hunk 5]
```diff
@@ -508,6 +509,22 @@ describe('ngb-popover', () => {
          expect(windowEl.textContent.trim()).toBe('Great tip!');
        }));
 
+    it('should modify the popper options', (done) => {
+      const fixture = createTestComponent(`<div ngbPopover="Great tip!" placement="right">Trigger</div>`);
+      const popover = fixture.debugElement.query(By.directive(NgbPopover)).injector.get(NgbPopover);
+
+      const spy = createSpy();
+      popover.popperOptions = (options: Partial<Options>) => {
```
Comment (maxokorokov) [inline src/popover/popover.spec.ts:517]: I wonder if we should explain how to use if more clearly in docs:
- is the initial config readonly or can be modified?
- should the user clone it?
- will initial config be discarded and ignored?

**Review by maxokorokov — CHANGES_REQUESTED** [review]
Hey, left some comments!

**Comment by fbasso** [comment]
Thanks for your review and comments, @maxokorokov !

I've changed the code accordingly.

**Review by maxokorokov — APPROVED** [review]
LGTM!

**Comment by sergeyglazyrindev** [comment]
Hello guys. Could you please suggest how do I proceed to get it working in my project ? The fix is not published yet on npm, right ? So, shall I use packag directly from github ? When are you gonna publish this version on npm ?

**Pull request merged**